src/view/scroll_view.rs

```diff
@@ -58,27 +62,22 @@ where
         id: &mut Id,
         state: &mut Self::State,
         element: &mut Self::Element,
-    ) -> bool {
+    ) -> ChangeFlags {
         cx.with_id(*id, |cx| {
             let child_element = element.child_mut().downcast_mut().unwrap();
-            let changed =
-                self.child
-                    .rebuild(cx, &prev.child, &mut state.0, &mut state.1, child_element);
-            if changed {
-                element.child_mut().request_update();
-            }
-            changed
+            self.child
+                .rebuild(cx, &prev.child, &mut state.0, &mut state.1, child_element)
         })
     }
 
-    fn event(
+    fn message(
         &self,
         id_path: &[Id],
         state: &mut Self::State,
         event: Box<dyn Any>,
         app_state: &mut T,
-    ) -> EventResult<A> {
+    ) -> MessageResult<A> {
         let tl = &id_path[1..];
-        self.child.event(tl, &mut state.1, event, app_state)
+        self.child.message(tl, &mut state.1, event, app_state)
```

Comment:
I know this is existing code, but I guess it makes sense to make this safer at this point.
(not sure how I can do a multiline suggestion across additions and existing code)

```suggestion
        match id_path {
            [child_id, rest_path @ ..] if *child_id == state.0 => {
                self.child
                    .message(rest_path, &mut state.1, message, app_state)
            }
            _ => MessageResult::Stale(message),
        }
```